### ISSUE TITLE
Sporadically list of spools empty or missing spools

### ISSUE THREAD

**Issue by Jeanlucpica** [OllisGit/OctoPrint-SpoolManager#236]
I have some strange behaviours with this plugin. Sporadically my list of spools will stay empty and show nothing, even though there should be several spools be displayed. Often it looks like this when I open the plugin page:
<img width="482" alt="2022-02-07 20_49_46-Prusa MINI+  OctoPrint  und 7 weitere Seiten - Persönlich – Microsoft​ Edge" src="https://user-images.githubusercontent.com/65825096/152861238-fe7fb768-e051-4f50-a007-1f8b7996234c.png">
A forced refresh usually then shows the complete list of spools.

But a more bothering issue happens especially after adding new spools or editing existing ones. As soon it goes back to the list suddenly some spools disappeared from the list. I can't really tell why some are shown afterwards and why some are not. Here is an example where 10 spools should actually be displayed, but I only see 7:
<img width="484" alt="2022-02-07 20_46_05-Prusa MINI+  OctoPrint  und 7 weitere Seiten - Persönlich – Microsoft​ Edge" src="https://user-images.githubusercontent.com/65825096/152861549-d0b510ec-cb82-4493-92ed-e616683aa704.png">

I already tried disabling some other plugins I have in use, but could not link this problem to a compatibility issue with another plugin. It seems to happen independently. Maybe somebody else has the same problem? Or can you give some advice on why this is happening?

Some technical details:
- Windows 10 (client)
- Microsoft Edge 98.0.1108.43 (client)
- OctoPrint 1.7.3
- OctoPi 0.18.0

**Comment by OllisGit** [comment]
Hi @Jeanlucpica,

regarding the first issue "empty list", pease take a look into the octoprint.log. Maybe the server could not load the spool infos from the database. Also, please open the "Browser-Developer-Tools (press F12)". In the console and on the Network-Tab there should be no red-lines.

In your screenshot is a color-filter active (show only 9 colors), maybe that is reason why you can't see all spools.
But I optimized the total-output in the next release.
![image](https://user-images.githubusercontent.com/24205767/154365438-634f3f9f-e98d-4fc6-ae32-a1db6231c8a6.png)

**Comment by OllisGit** [comment]
Hi @Jeanlucpica,

in the latest  RC-Verison  1.7.0rc1 I changed a lot of stuff, maybe this issue is also solved.
Please test and give feedback

Thx, in advance
Olli

**Comment by OllisGit** [comment]
Hi,
new Version 1.7.0 is released.

Please test and give feedback.

Thx, in advance
Olli

**Comment by Jeanlucpica** [comment]
Hi,
I installed version 1.7.0 and added a new spool. I can't really say it fixed things, but I believe I can give some more details what I am observing.

1. The list is always empty when switching to the "Spools" tab after a reboot of my OctoPi. If I hit the reload button everything comes up as expected. Not a problem, just inconvenient.
<img width="488" alt="image" src="https://user-images.githubusercontent.com/65825096/159121834-f86348ae-81ab-4e60-9509-87e5267b8c14.png">

2. I do still have some unexpected behaviour when adding/modifying a spool which led to the initial report here. I add a new spool like that:
<img width="746" alt="image" src="https://user-images.githubusercontent.com/65825096/159121867-a4ae5351-1407-4e5e-91e4-09796633b2a3.png">
First and last use are empty. I save the spool and I am back in the list. If I will now select a different spool which has been used before, it will show me all the details including the date for last and first use. Also fine. **But If I will now close that spool and then open the new spool, which was never used, it now will display the last and first use date of the spool I just opened before.** So it mixes up data. And if you do not pay attention and save it like this, you will save wrong data to the database.

So number 2 is an actual bug and I can reproduce it. It also happened to me that suddenly the color filter was set and excluded the color of the spool I just had added. However I am unable to reproduce it but it happened also with 1.7.0. Maybe you want to check if the problem of 2 maybe also applies to other fields and may cause this issue.

I hope this info helps to resolve this.
